lib/cached_counts.rb

```diff
@@ -138,34 +136,30 @@ def define_scope_count_attribute(attribute_name, relation, options)
       unless options.has_key?(:race_condition_fallback)
         options[:race_condition_fallback] = default_race_condition_fallback_proc(
           key,
-          relation,
           options
         )
       end
 
+      klass = self
+
       [attribute_name, *Array(options[:alias])].each do |attr_name|
         add_count_attribute_methods(
           attr_name,
           -> { key },
-          -> { relation },
+          -> { klass.send(scope_name) },
           :define_singleton_method,
           self,
           options
         )
       end
     end
 
-    def default_race_condition_fallback_proc(key, relation, options)
+    def default_race_condition_fallback_proc(key, options)
       fallback = Rails.cache.read(key)
       fallback = fallback.value if fallback.is_a?(ActiveSupport::Cache::Entry)
 
       if fallback.nil?
-        begin
-          fallback = relation.count
```

Comment:
This is where the actual query would be executed. We would only execute the query if the value was not found in the cahce